### PR TITLE
feat(notifier): disables provider selection when no active plans

### PR DESCRIPTION
**Pull request by itofarina**
## Features
- checks for any active plan in the providers to enable its selection

## Screenshot
Screen Shot 2021-07-13 at 16 36 30

### CODE REVIEW

File: src/components/pages/notifier/buy/NotifierOffersPage.tsx [hunk 1]
```diff
@@ -94,6 +94,8 @@ const NotifierOffersPage: FC = () => {
     .map<MarketplaceItem>((provider) => {
       const providerPlans = items.filter((item) => item.provider === provider)
 
+      const hasActivePlans = providerPlans.some(({ planStatus }) => planStatus === 'ACTIVE')
+
```
Comment (jurajpiar) [inline src/components/pages/notifier/buy/NotifierOffersPage.tsx]: Here, we need to instantiate each provider service and ask them directly using `SubscriptionPlans.hasPlans` method from `rif-notifier-seervice`.

```suggestion
      const notifierService = new SubscriptionPlans(url)
      notifierService.connect()
      const hasActivePlans = await notifierService.hasPlans()

```

Comment (jurajpiar) [reply src/components/pages/notifier/buy/NotifierOffersPage.tsx]: Basically, we are verifying that the provider is still serving plans.

**Review by jurajpiar — APPROVED** [review]
LGTM

File: src/components/pages/notifier/buy/NotifierOffersPage.tsx [hunk 2]
```diff
@@ -87,43 +90,71 @@ const NotifierOffersPage: FC = () => {
...
+          if (!hasActivePlans) {
+            action1 = (
+              <Tooltip title="This provider doesn't have any active plan at the moment">
+                <span>
+                  {selectButton}
+                </span>
+              </Tooltip>
+            )
+          }
```
Comment (jurajpiar) [inline src/components/pages/notifier/buy/NotifierOffersPage.tsx]: Mutation isn't nice thou...why not just
```suggestion
          const action1 = hasActivePlans
            ? selectButton
            :  (
              <Tooltip title="This provider doesn't have any active plan at the moment">
                <span>
                  {selectButton}
                </span>
              </Tooltip>
            )
```
?

Comment (itofarina) [reply src/components/pages/notifier/buy/NotifierOffersPage.tsx]: Done, thanks!

**Pull request merged**